File: 2325-decode-the-message/2325-decode-the-message.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        result = ""
        mapKeys = dict()
        asciiIndex = ord("a")
        for char in key:
            if len(mapKeys) == 26:
                break
            if char == " ":
                continue
            if char in mapKeys:
                continue

            asciiChar = chr(asciiIndex)
            mapKeys[char] = asciiChar
            asciiIndex += 1

        # print(mapKeys)
        for char in message:
            if char == " ":
                result += " "
                continue
            result += mapKeys[char]

        return result

File: 2325-decode-the-message/2325-decode-the-message.cs
public class Solution
{
  public string DecodeMessage(string key, string message)
  {
    var map = new Dictionary<char, char>();
    int counter = 97;
    foreach (char k in key)
    {
      if (map.ContainsKey(k) || k == ' ')
      {
        continue;
      }
      map.Add(k, Convert.ToChar(counter));
      counter++;
    }

    var result = new StringBuilder();
    foreach (char c in message)
    {
      result.Append(map.GetValueOrDefault(c, ' '));
    }
    return result.ToString();
  }
}